routing/turns_generator.cpp

```diff
@@ -668,6 +668,8 @@ size_t CheckUTurnOnRoute(TUnpackedPathSegments const & segments,
   auto const & masterSegment = segments[currentSegment - 1];
   if (masterSegment.m_path.size() < 2)
     return 0;
+  if (segments[currentSegment].m_path.empty())
```

Comment:
A few lines below:
...
auto const & checkedSegment = segments[currentSegment + i];
...
if (path[path.size() - 2] == checkedSegment.m_path[1])
...
Why you don't check that next segment's m_path is not empty before access 1st element?
May be need to add similar check?